### ISSUE TITLE
Issue: Problem with Using Melee and Certain Firearm Weapons in ox_inventory

### ISSUE THREAD

**Issue by xnooztvFR** [overextended/ox_inventory#1825]
**Describe the bug**
I am experiencing issues with certain weapons in the ox_inventory. When I attempt to equip melee weapons (such as knives or bats) and certain weapons (like weapon_smg_mk2), I receive an error message in the console stating that these weapons cannot be used in the current context (WEAPON_XXX cannot be used in current context (default game behaviour). As a result, the weapons are automatically removed from the inventory, preventing their usage.

**Framework**
ESX

**Resource version**
2.42.3

**To Reproduce**
1. Open the inventory.
2. Attempt to equip a melee weapon (e.g., knife, bat) or the weapon_smg_mk2.
3. See the error message in the console: "WEAPON_XXX cannot be used in current context (default game behaviour)".
4. The weapon gets removed from the inventory and cannot be used.

**Expected behavior**
I expected to be able to equip and use the melee weapons and the weapon_smg_mk2 without any issues.

**Screenshots**
Currently, I do not have screenshots to provide.

**Additional context**
I have verified that other weapons (like pistols and rifles) work correctly without issues. The affected weapons are defined correctly in the database. If there are any configurations or settings that I need to check, please advise.

**Comment by thelindat** [comment]
> the weapons are automatically removed from the inventory

?????????

This error is literally just the weapon either being forcibly removed **by the game** or failing to be equipped in the first place because **they cannot be used, per the game**. There's nothing deleting the items or explicitly disabling those items as part of the resource.

**Comment by xnooztvFR** [comment]
Indeed, the weapon is not removed from the inventory, but it is holstered in the inventory. However, I don't understand why this happens with certain weapons (for example, WEAPON_KNIFE and WEAPON_SMG_MK2). Is there a solution?"


> > the weapons are automatically removed from the inventory
> 
> ?????????
> 
> This error is literally just the weapon either being forcibly removed **by the game** or failing to be equipped in the first place because **they cannot be used, per the game**. There's nothing deleting the items or explicitly disabling those items as part of the resource.
> 
> https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L529-L535
> 
> https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L1388-L1406

**Comment by Lymeish** [comment]
> Indeed, the weapon is not removed from the inventory, but it is holstered in the inventory. However, I don't understand why this happens with certain weapons (for example, WEAPON_KNIFE and WEAPON_SMG_MK2). Is there a solution?"
> 
> > > the weapons are automatically removed from the inventory
> > 
> > 
> > ?????????
> > This error is literally just the weapon either being forcibly removed **by the game** or failing to be equipped in the first place because **they cannot be used, per the game**. There's nothing deleting the items or explicitly disabling those items as part of the resource.
> > https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L529-L535
> > 
> > https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L1388-L1406

Are you using add on weapons?

Ensure your correctly installed the weapons stream/data/meta files or you get this error.

**Comment by xnooztvFR** [comment]
> > Indeed, the weapon is not removed from the inventory, but it is holstered in the inventory. However, I don't understand why this happens with certain weapons (for example, WEAPON_KNIFE and WEAPON_SMG_MK2). Is there a solution?"
> > > > the weapons are automatically removed from the inventory
> > > 
> > > 
> > > ?????????
> > > This error is literally just the weapon either being forcibly removed **by the game** or failing to be equipped in the first place because **they cannot be used, per the game**. There's nothing deleting the items or explicitly disabling those items as part of the resource.
> > > https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L529-L535
> > > 
> > > https://github.com/overextended/ox_inventory/blob/c0abafb246ecfababc18c7f8fe04abf2fcfdfd6f/client.lua#L1388-L1406
> 
> Are you using add on weapons?
> 
> Ensure your correctly installed the weapons stream/data/meta files or you get this error.

No I don't have any weapons addons. I tried with only ox_inventory and the problem is still there.

**Comment by Restitive** [comment]
Got the same error. Is there some fix?
